### PR TITLE
feat: migrate Redshift settings forms to use Mantine

### PR DESCRIPTION
**Pull request by jayanand05**
Closes: This PR closes one task from Warehouse connection (i.e Redshift) defined in the issue #5402 

### Description:

Updating Redshift defined in WarehouseConnection inside Project Settings

### CODE REVIEW

File: packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx [hunk 1]
```diff
@@ -70,183 +76,245 @@ const RedshiftForm: FC<{
...
+                                />
+                            )}
+                        />
+
+                        <BooleanSwitch
+                            name="warehouse.ra3Node"
+                            label="Use RA3 node"
+                            labelHelp="Allow dbt to use cross-database-resources."
+                            defaultValue
+                            disabled={disabled}
```
Comment (jayanand05) [inline packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:217]: just wanted to ask can we use below code in place of BooleanSwitch defined above `       <Controller
                            name="warehouse.ra3Node"
                            render={({ field }) => (
                                <Switch.Group
                                    label="Use RA3 node"
                                    description="Allow dbt to use cross-database-resources."
                                    value={field.value}
                                    onChange={field.onChange}
                                >
                                    <Switch
                                        onLabel="Yes"
                                        offLabel="No"
                                        {...field}
                                        disabled={disabled}
                                        value="true"
                                        onChange={(isChecked) =>
                                            field.onChange(isChecked)
                                        }
                                    />
                                </Switch.Group>
                            )}
                        />`

File: packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx [hunk 2]
```diff
@@ -70,183 +76,245 @@ const RedshiftForm: FC<{
...
+                                    onChange={field.onChange}
+                                    disabled={disabled}
+                                />
+                            )}
+                        />
+
+                        <BooleanSwitch
+                            name="warehouse.ra3Node"
+                            label="Use RA3 node"
+                            labelHelp="Allow dbt to use cross-database-resources."
```
Comment (jayanand05) [inline packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:215]: just wanted to ask can we use below code in place of BooleanSwitch defined above <Controller name="warehouse.ra3Node" render={({ field }) => ( <Switch.Group label="Use RA3 node" description="Allow dbt to use cross-database-resources." value={field.value} onChange={field.onChange} > <Switch onLabel="Yes" offLabel="No" {...field} disabled={disabled} value="true" onChange={(isChecked) => field.onChange(isChecked) } /> </Switch.Group> )} />

File: packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx [hunk 3]
```diff
@@ -70,183 +76,245 @@ const RedshiftForm: FC<{
...
+                            defaultValue
```
Comment (jayanand05) [inline packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:216]: just wanted to ask can we use below code in place of BooleanSwitch defined above
`       <Controller
                            name="warehouse.ra3Node"
                            render={({ field }) => (
                                <Switch.Group
                                    label="Use RA3 node"
                                    description="Allow dbt to use cross-database-resources."
                                    value={field.value}
                                    onChange={field.onChange}
                                >
                                    <Switch
                                        onLabel="Yes"
                                        offLabel="No"
                                        {...field}
                                        disabled={disabled}
                                        value="true"
                                        onChange={(isChecked) =>
                                            field.onChange(isChecked)
                                        }
                                    />
                                </Switch.Group>
                            )}
                        />`

File: packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx (hunk 2 above)
Comment (jayanand05) [inline packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:215]: just wanted to ask can we use below code in place of BooleanSwitch defined above
`       <Controller
                            name="warehouse.ra3Node"
                            render={({ field }) => (
                                <Switch.Group
                                    label="Use RA3 node"
                                    description="Allow dbt to use cross-database-resources."
                                    value={field.value}
                                    onChange={field.onChange}
                                >
                                    <Switch
                                        onLabel="Yes"
                                        offLabel="No"
                                        {...field}
                                        disabled={disabled}
                                        value="true"
                                        onChange={(isChecked) =>
                                            field.onChange(isChecked)
                                        }
                                    />
                                </Switch.Group>
                            )}
                        />`

**Review by ZeRego — APPROVED** [review]

File: packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx [hunk 4]
```diff
@@ -70,183 +77,278 @@ const RedshiftForm: FC<{
...
+                            name="warehouse.ra3Node"
+                            render={({ field }) => (
+                                <Switch.Group
+                                    label="Use RA3 node"
+                                    description="Allow dbt to use cross-database-resources."
+                                    value={field.value ? ['true'] : []}
+                                    onChange={(values) =>
+                                        field.onChange(values.length > 0)
+                                    }
+                                    size="md"
```
Comment (jayanand05) [inline packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:223]: Hi @ZeRego , do we need `size="md"` here?

Comment (ZeRego) [reply packages/frontend/src/components/ProjectConnection/WarehouseForms/RedshiftForm.tsx:223]: We don't need it but I felt it looked slightly better.
With
Screenshot 2023-09-21 at 14 56 15
Without
Screenshot 2023-09-21 at 14 56 02

**Pull request merged**